zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/elasticsearch/ElasticsearchEngineClient.java

```diff
@@ -251,6 +255,32 @@ public void updateIndexTemplateSettings(
     }
   }
 
+  @Override
+  public void deleteIndex(final String indexName) {
+    final DeleteIndexRequest deleteIndexRequest =
+        new DeleteIndexRequest.Builder().index(indexName).build();
+    try {
+      final AcknowledgedResponse deleteIndexResponse = client.indices().delete(deleteIndexRequest);
+      LOG.debug("Delete index acknowledged: {}", deleteIndexResponse.acknowledged());
+    } catch (final IOException e) {
+      final var errMsg = String.format("Failed to delete index %s", indexName);
+      LOG.error(errMsg, e);
+    }
+  }
+
+  @Override
+  public void truncateIndex(final String indexName) {
+    final DeleteByQueryRequest deleteByQueryRequest =
+        new DeleteByQueryRequest.Builder().index(indexName).query(q -> q.matchAll(m -> m)).build();
+    try {
+      final DeleteByQueryResponse response = client.deleteByQuery(deleteByQueryRequest);
+      LOG.debug("Deleted {} documents from index {}", response.deleted(), indexName);
+    } catch (final IOException e) {
+      final var errMsg = String.format("Failed to delete docs from index %s", indexName);
+      LOG.error(errMsg, e);
```

Comment:
:x: Are we not rethrowing? Won't that mean we can "succeed" purging but there was an error? Could elaborate?

Reply:
Looking at the OpenSearch implementation where we rethrow - I see a couple things.

1. If we're rethrowing (which I think is the right behavior), then we should not log as error here. The error will be logged by the caller. Right now it's likely just redundant (do verify though, I could be wrong).
2. We anyway typically do not log recoverable errors (which IOException tend to be) as error, since the idea is they are transient. We'd log them as warnings. Error should be reserved for those which require immediate action.